core/src/main/java/io/questdb/griffin/WhereClauseParser.java

```diff
@@ -1158,6 +1180,42 @@ private boolean removeWithin(AliasTranslator translator, ExpressionNode node, Re
         }
     }
 
+    private void processArgument(
+            ExpressionNode inArg,
+            RecordMetadata metadata,
+            FunctionParser functionParser,
+            SqlExecutionContext executionContext,
+            int columnType,
+            LongList prefixes
+    ) throws SqlException, NumericException {
+        if(isNull(inArg)) {
+            throw SqlException.$(inArg.position, "GeoHash value expected");
+        }
+        if (isFunc(inArg)) {
+            Function f = functionParser.parseFunction(inArg, metadata, executionContext);
+            if (isGeoHashConstFunction(f)) {
+                final int fnType = f.getType();
+                final long hash = f.getGeoHashLong(null);
+                GeoHashes.addNormalizedGeoPrefix(hash, fnType, columnType, prefixes);
+            } else {
+                throw SqlException.$(inArg.position, "GeoHash const function expected");
+            }
+        } else {
+            if (inArg.type != ExpressionNode.CONSTANT) {
+                throw SqlException.$(inArg.position, "GeoHash literal expected");
+            }
+            GeoHashes.addNormalizedGeoPrefix(unquote(inArg.token), columnType, prefixes);
+        }
+    }
+
+    private boolean isNull(ExpressionNode node) {
+        return node == null || isNullKeyword(node.token);
+    }
+
+    private boolean isGeoHashConstFunction(Function fn) {
+        return (fn instanceof GeoHashFunction) && (fn.isConstant() || fn.isRuntimeConstant());
```

Comment:
You cannot get value of Runtime Constant at compilation time. Example of Runtime Constant function is now() which is different every execution even when CursorFactory is reused